### ISSUE TITLE
Docker build for armhf architecture: Dockerfile changes

### ISSUE THREAD

**Issue by matieux** [borgmatic-collective/docker-borgmatic#54]
Hi, good stuff.

Problem: The base Docker image build doesn't work for Linux Alpine 3.12 on `armhf` architecture. The `py3-wheel` package is not available and only `python3.6` is installed. Below is a working `Dockerfile`.

I can make a pull request, just let me know how to organize the project folders per architecture.

```docker
FROM alpine:latest as builder
LABEL mainainer='b3vis'
ARG BORG_VERSION=1.1.14
ARG BORGMATIC_VERSION=1.5.10
ARG LLFUSE_VERSION=1.3.6
RUN apk upgrade --no-cache \
    && apk add --no-cache \
    alpine-sdk \
    python3-dev \
    py3-pip \
    openssl-dev \
    lz4-dev \
    acl-dev \
    linux-headers \
    fuse-dev \
    attr-dev
#    py3-wheel \
RUN pip3 install --upgrade pip \
    && pip3 install --upgrade borgbackup==${BORG_VERSION} \
    && pip3 install --upgrade borgmatic==${BORGMATIC_VERSION} \
    && pip3 install --upgrade llfuse==${LLFUSE_VERSION}

FROM alpine:latest
LABEL mainainer='b3vis'
COPY entry.sh /entry.sh
RUN apk upgrade --no-cache \
    && apk add --no-cache \
    tzdata \
    sshfs \
    python3 \
    openssl \
    fuse \
    ca-certificates \
    lz4-libs \
    libacl \
    postgresql-client \
    mariadb-client \
    && rm -rf /var/cache/apk/* \
    && chmod 755 /entry.sh
VOLUME /mnt/source
VOLUME /mnt/borg-repository
VOLUME /etc/borgmatic.d
VOLUME /root/.borgmatic
VOLUME /root/.config/borg
VOLUME /root/.ssh
VOLUME /root/.cache/borg
#COPY --from=builder /usr/lib/python3.8/site-packages /usr/lib/python3.8/
COPY --from=builder /usr/lib/python3.6/site-packages /usr/lib/python3.6/
COPY --from=builder /usr/bin/borg /usr/bin/
COPY --from=builder /usr/bin/borgfs /usr/bin/
COPY --from=builder /usr/bin/borgmatic /usr/bin/
COPY --from=builder /usr/bin/generate-borgmatic-config /usr/bin/
COPY --from=builder /usr/bin/upgrade-borgmatic-config /usr/bin/
CMD ["/entry.sh"]
```

**Comment by b3vis** [comment]
Hi!

Thanks for the information I had been thinking for a while about getting a ARM architecture image for this. A PR would be most welcomed, I think if you could fork and create a branch to PR against that would be great.

As for structure I'm not 100% on how Dockerhub looks for different build image files but a hunch says naming them Dockerfile.armhf should be sufficient 

Thanks!

b3vis

**Comment by b3vis** [comment]
@matieux - Hey any progress on this?

**Comment by MarkJonas** [comment]
A have a multi-platform Docker build running.

https://gitlab.com/toertel/docker-image-logitech-media-server

If there are some questions I would happy to help out.

To avoid duplication of code in the Dockerfiles I recommend the usage of j2cli (Jinja2 CLI).

**Comment by frutti93** [comment]
Hi everyone, i recently forked this repo because i wanted arm64 builds. So I created a github workflow to build the image for multiple architectures. I would be happy to create a PR if you are interested (@b3vis) ? 

https://github.com/frutti93/docker-borgmatic

**Comment by b3vis** [comment]
@frutti93 yes please definitely interested in a PR for that

**Comment by frutti93** [comment]
Created the PR

**Comment by modem7** [comment]
I've forked a multi-arch version of Borgmatic if you wish to have a look

https://hub.docker.com/r/modem7/borgmatic-docker

**Comment by frutti93** [comment]
Thank, I am currently also building the images myself, but as this repo is more popular than mine i thought implementing this here could not hurt either. :)

**Comment by modem7** [comment]
Absolutely agreed @frutti93!

For me, I wanted docker commands to be able to be executed for backup purposes so I could stop/start containers, so it's easier to run a fork in my particular case

**Comment by b3vis** [comment]
Thanks for the PR, do you have a write up on how to implement this? I haven't used GitHub actions before

**Comment by frutti93** [comment]
Basically you only have to add the variables DOCKER_USERNAME and DOCKER_PASSWORD to your repository secrets after merging the PR.
Github actions works by reading from the files in a .github directory in your repository, as you can see in my PR.
They will be triggered by a push to the master branch or by creating a new tag.
Then the docker images are built and tagged by Github and pushed to your Dockerhub.

You can see the commands which are executed to do this in the files I provided. ;)

**Comment by PixelJonas** [comment]
@frutti93 is the arm version you are building public somewhere? I would like to give it a spin

**Comment by frutti93** [comment]
@PixelJonas I do not have the "pure" version built, but I could offer you my fork. It is basically the same, I just included an optional rclone hook because I am uploading my backup to Backblaze after the backup has completed:

https://github.com/frutti93/docker-borgmatic

**Comment by modem7** [comment]
> @PixelJonas I do not have the "pure" version built, but I could offer you my fork. It is basically the same, I just included an optional rclone hook because I am uploading my backup to Backblaze after the backup has completed:
> 
> https://github.com/frutti93/docker-borgmatic
> 
> If you would like I could also just build a clean fork of this repo until my PR is completed.

@frutti93 - regarding adding additional things to the image, I've done the following: https://github.com/modem7/docker-borgmatic/blob/master/base-liveinstall/entry.sh

Might be useful if you want to create an image that loads (in this case rclone) when the container is started, keeping the base image effectively virgin.

**Comment by b3vis** [comment]
Apologies all, this has now been implemented, I have also just released v1.1.17-1.5.21 which is building through GitHub actions now. I have also applied for a Docker Hub OSS Org to move it from my personal Docker Hub account to borgmatic-collective/borgmatic 👍🏽
